src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/xresilience/RetriesSpec.groovy

```diff
@@ -261,22 +267,22 @@ and at least 1 path must remain safe"
     def "System retries the intentional rerouteV1 if it fails to install rules on a switch"() {
         given: "Two active neighboring switches with two diverse paths at least(main and backup paths)"
         def swPair = switchPairs.all().neighbouring().withAtLeastNNonOverlappingPaths(2).random()
+        def availablePaths = swPair.retrieveAvailablePaths().collect { it.getInvolvedIsls() }
 
-        List<PathNode> mainPath = swPair.paths.min { it.size() }
+        List<Isl> mainPathIsls = availablePaths.min { it.size() }
         //find path with more than two switches
-        List<PathNode> backupPath = swPair.paths.findAll { it != mainPath && it.size() != 2 }.min { it.size() }
+        List<Isl> backupPathIsls = availablePaths.findAll { it != mainPathIsls && it.size() > 1 }.min { it.size() }
```

Comment:
As we use Isl path representation: more that 1 ISL is 2 and more switches in path